robot/session_manager.go

```diff
@@ -20,7 +20,7 @@ func NewSessionManager(robot Robot, heartbeatWindow time.Duration) *SessionManag
 	m := &SessionManager{
 		robot:             robot,
 		heartbeatWindow:   heartbeatWindow,
-		logger:            logging.FromZapCompatible(robot.Logger().Named("session_manager")),
+		logger:            robot.Logger().Sublogger("session_manager"),
```

Comment:
Now that we have a `Sublogger` method, we can keep a viam-logger -> viam-logger without losing information passing through a zap logger. A test failure caught this.

The "fix" to copy the observer cores from a viam-logger -> zap logger also would have fixed the test failure this change also fixes. But the goal is to only use zap when interfacing with other repositories. There may be other cases, but those will get cleaned up when changing viam in production to use the viam logger.